File: 1256-rank-transform-of-an-array/1256-rank-transform-of-an-array.cs
public class Solution {
    public int[] ArrayRankTransform(int[] arr) {
        int[] sortedArr = (int[])arr.Clone();
        Array.Sort(sortedArr);

        Dictionary<int, int> rankMap = new Dictionary<int, int>();
        int rank = 1;

        foreach(int num in sortedArr) {
            if(!rankMap.ContainsKey(num)) {
                rankMap[num] = rank;
                rank += 1;
            }
        }

        int[] rankedArr = new int[arr.Length];

        for (int i = 0; i < arr.Length; i++) {
            rankedArr[i] = rankMap[arr[i]];
        }

        return rankedArr;
    }
}

File: 1256-rank-transform-of-an-array/1256-rank-transform-of-an-array.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        map<int,int> hash;

        for(auto num : arr) 
            hash[num] = 0;
        
        int rank = 1;
        for(auto& num : hash) {
            num.second = rank;
            rank+= 1;
        }

        vector<int> rankedArr;
        for(auto num : arr) {
            rankedArr.push_back(hash[num]);
        }

        return rankedArr;
            
    }
};